Expanded Security Advisor blog to describe Exporter

KalmanMeth opened the pull request:
Added introductory paragraph to the Explorer and added a table with all the defined fields.

KalmanMeth commented:
@hunchback @eranra For your review.

eranra commented on blog/exporters.md:
```diff
@@ -1,19 +1,22 @@
 ---
-title: How to use Security Advisor with Skydive
+title: How to use Exporter pipeline with Skydive
```
Change to flow exporter

eranra commented on blog/exporters.md:
Why "secadvisor" in the path ... I thought we move secadvisor to be only configuration ... in any event the README.md should moveto https://github.com/skydive-project/skydive/tree/master/contrib/exporters/ level @hunchback ?

hunchback replied:
for now the README.md is specific to secadvisor, but I agree - it should be generalized and moved one level up in the future (and then we can update the link here)

eranra commented on blog/exporters.md:
```diff
@@ -60,9 +63,10 @@ sudo skydive agent -c /etc/skydive/skydive.yml
 
-Start from the `secadvisor.yml.default` in the secadvisor directory.
+Start from the `secadvisor.yml.default` in the `secadvisor` directory.
+Different applications can be built up by specifying different parameters in the configuration file.
```
Same here ... I expect to have flow_exporter.yml ... and secadvisor.yml.default can be additional one ... but this doc should point to the generic one and not to secadvisor.

hunchback replied:
Again - for now it should still point to `secadvisor.yml.default` as the blog must accurately represent the current state - in future  based on changes we make to the code base (such as moving to use `uber` which includes also the secadvisor specific phases) we can adapt this blog

eranra reviewed (commented):
Also work with @hunchback to add information on (1) VPC logs and (2) AWS flow export data ... so I expect in this blog to see three concrete use-cases / exporter flows pipe line based on the generic one ... 

(I really like the table :-)  )

hunchback commented on blog/exporters.md:
```diff
@@ -1,19 +1,22 @@
 
+The Exporter filters the flow data obtained from Skydive, performs a data transformation, and saves the information to some target (e.g. an object store).
+This data may then be used to perform various kinds of analyses for security, accounting, or other purposes.
+We use the term `pipeline` to describe the sequence of operations performed on the data: `classify`, `filter`, `transform`, `encode`, `store`.
```
add `compress` after `transform`

KalmanMeth replied:
OK

hunchback commented on blog/exporters.md:
I suggest you remove this line

KalmanMeth replied:
OK

hunchback commented on blog/exporters.md:
Rather write:

```
We also have a Security Advisor exporter which feeds the Secutiy Advisor application with flow data captured using Skydive.
```

KalmanMeth replied:
The sentence was removed and replaced with
"We describe below the use of the Flow Exporter for the Security Advisor application."

hunchback replied:
ok

hunchback commented on blog/exporters.md:
Rewrite to:

```
The modeling of the exporter as a multi-phase pipeline, means that its' possible better reuse specific phases while implementing new exporters.
```

KalmanMeth replied:
rewritten

hunchback commented on blog/exporters.md:
````diff
@@ -74,48 +78,102 @@ The `secadvisor.yml` is used as a parameter when running secadvisor from the com
 The default `secadvisor.yml` has the following fields:
 
 ```
+# host_id is used to reference the agent, should be unique or empty
````
host_id is used to identify the webscoket client application, should be unique or empty

KalmanMeth replied:
OK

hunchback commented on blog/exporters.md:
```diff
@@ -74,48 +78,102 @@ The `secadvisor.yml` is used as a parameter when running secadvisor from the com
+| analyzers             | URL            | address of Skydive analyzer for queries |
+| pipeline              | list of params | pipeline specification |
+| analyzer              | list of params | analyzer credentials for websocket connection |
+| subscriber_url        | URL            | websocket address to obtain flows from Skydive |
```
give full path e.g. `pipeline.analyzer.subscriber_url`

KalmanMeth replied:
OK

hunchback commented on blog/exporters.md:
```diff
@@ -74,48 +78,102 @@ The `secadvisor.yml` is used as a parameter when running secadvisor from the com
+| subscriber_username   | string         | websocket creedentials |
+| subscriber_password   | string         | websocket creedentials |
+| cluster_net_masks     | list of cidr   | list of net-addr/net_mask to be considered internal addresses |
+| filter                | list of params | filter options; currently the only parameter is `excluded_tags` |
```
redundant stating that filter has excluded_tags here when a line below you have an entry for excluded_tags

KalmanMeth replied:
At present it is redundant because we have only one option. If we don't expect any future changes, then these could be combined into a single entry. The fact that we have them separated enables the possibility of adding future suboptions.

hunchback replied:
I expect more options in future, but no need that at the top-level `filter` description to repeat the suboptions.

hunchback commented on blog/exporters.md:
```diff
@@ -74,48 +78,102 @@ The `secadvisor.yml` is used as a parameter when running secadvisor from the com
+| endpoint              | URL            | address of object store |
+| region                | string         | used to access aws `region` |
+| bucket                | string         | name of bucket in object store where data will be stored |
+| access_key            | string         | credentials for aws-type object store |
```
provide a link to AWS documentation explaining this parameter

KalmanMeth replied:
Added to later section where parameters are elaborated upon.

hunchback replied:
ok

hunchback commented on blog/exporters.md:
```diff
@@ -74,48 +78,102 @@ The `secadvisor.yml` is used as a parameter when running secadvisor from the com
+| region                | string         | used to access aws `region` |
+| bucket                | string         | name of bucket in object store where data will be stored |
+| access_key            | string         | credentials for aws-type object store |
+| secret_key            | string         | credentials for aws-type object store |
```
same

KalmanMeth replied:
OK

hunchback commented on blog/exporters.md:
```diff
@@ -74,48 +78,102 @@ The `secadvisor.yml` is used as a parameter when running secadvisor from the com
+| bucket                | string         | name of bucket in object store where data will be stored |
+| access_key            | string         | credentials for aws-type object store |
+| secret_key            | string         | credentials for aws-type object store |
+| api_key               | string         | credentials for IBM COS-type object store |
```
same

KalmanMeth replied:
This is described in the text after the table where there is more elaboration on the keys.

hunchback replied:
maybe embed the below text into the table ... it may be combustion to have both the table and the longer explanation below

hunchback commented on blog/exporters.md:
```diff
@@ -74,48 +78,102 @@ The `secadvisor.yml` is used as a parameter when running secadvisor from the com
+| access_key            | string         | credentials for aws-type object store |
+| secret_key            | string         | credentials for aws-type object store |
+| api_key               | string         | credentials for IBM COS-type object store |
+| iam_endpoint          | string         | credentials for IBM COS-type object store |
```
same

KalmanMeth replied:
same

hunchback commented on blog/exporters.md:
```diff
@@ -176,18 +234,18 @@ In the `secadvisor.yml` file, uncomment the `iam_endpoint` field and set it to `
 For running tests on a local machine, it is possible to set up a local Minio object store.
 
-For details, see the instructions found at [Secadvisor Developer's README](https://github.com/skydive-project/skydive/tree/master/contrib/pipelines/secadvisor/README.md).
+For details, see the instructions found at [Secadvisor Developer's README](https://github.com/skydive-project/skydive/tree/master/contrib/exporters/secadvisor/README.md).
```
You have already referenced this README above ... not sure you need to reference it again throughout this blog

hunchback commented on blog/exporters.md:
```diff
@@ -176,18 +234,18 @@ In the `secadvisor.yml` file, uncomment the `iam_endpoint` field and set it to `
 ## Deploy the Security Advisor pipeline
 
-Build and run the pipeline in the secadvisor directory: $SKYDIVE_BASE/contrib/pipelines/secadvisor.
+Build and run the pipeline in the secadvisor directory: $SKYDIVE_BASE/contrib/exporters/secadvisor.
```
s/pipeline/exporter/

KalmanMeth replied:
I thought pipeline directory is being removed from the path. The current path of the source is $SKYDIVE_BASE/contrib/exporters/secadvisor.

hunchback replied:
I meant fix:

```
Build and run the pipeline in the secadvisor directory: 
```

To:

```
Build and run the exporter in the secadvisor directory: 
```

KalmanMeth replied:
OK

hunchback commented on blog/index.md:
```diff
@@ -3,12 +3,12 @@ title: Blog
-The Security Advisor filters the flow data obtained from Skydive, performs a data transformation, and saves the information to an object store in JSON format. This data may then be used to perform various kinds of analyses for security, accounting, or other purposes.
-In this article, we give full instructions to install and use the Security Advisor.
+Support was recently added for an Exporter add-on to Skydive.
+The Exporter filters the flow data obtained from Skydive, performs a data transformation, and saves the information to an object store. This data may then be used to perform various kinds of analyses for security, accounting, or other purposes.
```
s/object store/storage target (e.g. AWS-S3)/

KalmanMeth replied:
OK

hunchback commented:
@KalmanMeth - please let me know when you push the final version and are ready for a review

hunchback commented:
@KalmanMeth - please squash commits

hunchback commented on blog/exporters.md:
```diff
@@ -365,6 +431,37 @@ UpdateCount - number of updates in this report.
+  sa:
+    exclude_started_flows: false
+    extend:
+      - VAR_NAME1=<gremlin expression with substitution strings>
```
<gremlin expression in go template format to enable field substitution>

KalmanMeth committed:
Expanded Security Advisor blog to describe Exporter and vpclogs

KalmanMeth commented:
@eranra @hunchback 
Documentation for the Flow Exporter is updated, including details of vpclogs.

hunchback commented:
I think there is just too much content in this blog, it was long before, and now its' become even longer. 

My suggestion is that you **create a new blog**  describing the exporter (note the top-level readm PR https://github.com/skydive-project/skydive-flow-exporter/blob/6c5e8927a7260844aa8fd6ceb4602ff746586ce4/README.md)

KalmanMeth committed:
Separated out details of Security Advisor and VPC logs details from Flow Exportter blog

KalmanMeth commented:
@hunchback 
As recommended, the blog now concentrates on the Flow Exporter options, and the details for Security Advisor and VPC logs are separated out in their own pages.

KalmanMeth commented:
@eranra @safchain @lebauce FYI I updated here the blog that describes the configuration parameters for the skydive flow exporter and addressed Aidan's last comment that it was too long. The details for the Security Advisor and VPC logs were pulled out into separate pages, and the current main page concentrates on the config parameters for the flow exporter itself.

safchain commented:
There are few errors reported:
```
jekyll build
Ignoring commonmarker-0.17.13 because its extensions are not built. Try: gem pristine commonmarker --version 0.17.13
Ignoring ffi-1.12.2 because its extensions are not built. Try: gem pristine ffi --version 1.12.2
Configuration file: /home/safchain/code/skydive.network/_config.yaml
            Source: /home/safchain/code/skydive.network
       Destination: /home/safchain/code/skydive.network/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
    Liquid Warning: Liquid syntax error (line 378): [:end_of_string] is not a valid expression in "{{ }}" in blog/secadvisor.md
    Liquid Warning: Liquid syntax error (line 379): [:end_of_string] is not a valid expression in "{{ }}" in blog/secadvisor.md
    Liquid Warning: Liquid syntax error (line 383): [:dot, "."] is not a valid expression in "{{.Network.A}}" in blog/secadvisor.md
    Liquid Warning: Liquid syntax error (line 384): [:dot, "."] is not a valid expression in "{{.Network.B}}" in blog/secadvisor.md
                    done in 1.131 seconds.
 Auto-regeneration: disabled. Use --watch to enable.
```

I would also change the publication date and order in the index file

KalmanMeth committed:
fixed error found by jekyll
